File: 1006.scala
object Main {

    def main(args: Array[String]) {

        var x = scala.io.StdIn.readDouble()
        var y = scala.io.StdIn.readDouble()
        var z = scala.io.StdIn.readDouble()
        x = x * 2 + y * 3 + z * 5
        x /= 10.0
        
        println(f"MEDIA = $x%.1f")
    }

}

File: solutions/1006.rs
use std::{io, process::exit};

// Read integer
fn read_int() -> i32 {
  let mut input_text = String::new();
  io::stdin()
    .read_line(&mut input_text)
    .expect("Deu merda");

  let trimmed = input_text.trim();
  match trimmed.parse::<i32>() {
    Ok(i) => return i,
    Err(..) => exit(1),
  };
}

// Read float
fn read_float() -> f64 {
  let mut input_text = String::new();
  io::stdin()
    .read_line(&mut input_text)
    .expect("Deu merda");

  let trimmed = input_text.trim();
  match trimmed.parse::<f64>() {
    Ok(i) => return i,
    Err(..) => exit(1),
  };
}

fn main() {

  // :)
  let a = read_float();
  let b = read_float();
  let c = read_float();
  println!("MEDIA = {:.1}", (a * 2.0 + b * 3.0 + c * 5.0) / 10.0);
}
